src/main/java/gift/controller/MemberController.java

```diff
@@ -48,4 +57,34 @@ public ResponseEntity<Map<String, String>> login(@RequestBody MemberRequest memb
                 .body(responseBody);
     }
 
+    @GetMapping("/kakao/login")
+    public ResponseEntity<Map<String, String>> kakaoCallback(@RequestParam String code) {
+        KakaoTokenResponse tokenResponse = kakaoLoginService.getKakaoToken(code);
+
+        KakaoProfileResponse profileResponse = kakaoLoginService.getUserProfile(tokenResponse.accessToken());
+
+        String email = profileResponse.kakaoAccount().profile().nickname() + "@kakao.com";
+
+        if (email.equals("@kakao.com")) {
```

Comment:
email 이 Null 일 가능성은 없어보이지만, 상수인 "@kakao.com" 으로 비교하는 것도 Npe 방지 목적에서 안전해보여요.
```suggestion
        if ("@kakao.com".equals(email)) {
```